### PR TITLE
R21b do not merge

### PR DESCRIPTION
**Pull request by its-pointless**
i need to integrate libllvm 10 and test rust to make sure this is smooth and we don't waste bandwidth.

### CODE REVIEW

**Comment by buttaface** [comment]
Let me know if you have any problem with the LLVM 10 pull. I used this patch to build Rust with it and it seemed to be working fine, only to run out of disk space on the CI after building LLVM somewhere near the end of building Rust:
```
diff --git a/packages/rust/build.sh b/packages/rust/build.sh
index f67a024fb..d8629d6fd 100644
--- a/packages/rust/build.sh
+++ b/packages/rust/build.sh
@@ -42,7 +42,7 @@ termux_step_configure() {
 	export CFLAGS_x86_64_unknown_linux_gnu="-O2"
 	unset CC CXX CPP LD CFLAGS CXXFLAGS CPPFLAGS LDFLAGS PKG_CONFIG AR RANLIB
 	# we can't use -L$PREFIX/lib since it breaks things but we need to link against libLLVM-9.so
-	ln -sf $PREFIX/lib/libLLVM-9.0.1.so $TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/
+	ln -sf $PREFIX/lib/libLLVM-10.0.0.so $TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/
 
 	# rust checks libs in PREFIX/lib because both host and target are x86_64. It then can't find libc.so and libdl.so because rust program doesn't 
 	# know where those are. Putting them temporarly in $PREFIX/lib prevents that failure
@@ -80,7 +80,7 @@ termux_step_make_install() {
 		rust-installer-version \
 		manifest-* \
 		x86_64-unknown-linux-gnu
-	rm $TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/libLLVM-9.0.1.so 
+	rm $TERMUX_STANDALONE_TOOLCHAIN/sysroot/usr/lib/$TERMUX_HOST_PLATFORM/$TERMUX_PKG_API_LEVEL/libLLVM-10.0.0.so
 
 }
 termux_step_post_massage() {
```

File: scripts/setup-ubuntu.sh [hunk 1]
```diff
@@ -106,9 +107,6 @@ PACKAGES+=" zlib1g-dev"
 PACKAGES+=" libssl-dev:i386"
 PACKAGES+=" zlib1g-dev:i386"
 
-# So we don't build llvm for build.
-PACKAGES+=" llvm-8-tools"
-
 # For swift.
 PACKAGES+=" clang-9"
```
Comment (buttaface) [inline scripts/setup-ubuntu.sh:111]: Now that you added clang-10 above, you can remove this line and pull in my updated Swift commit that uses the newer clang, buttaface/termux-packages@222f67f.

Comment (its-pointless) [reply scripts/setup-ubuntu.sh:111]: okay

**Comment by buttaface** [comment]
You can remove the last Swift commit: I submitted a separate pull for that, so it's already in.

**Comment by buttaface** [comment]
Hey, just noticed that [Carmack tweeted about Termux](https://mobile.twitter.com/ID_AA_Carmack/status/1264682608857952259), you all should be proud. :smile:

**Comment by Grimler91** [comment]
So, this is ready to be merged (once merge conflicts are resolved), right? (Even though title says "do not merge").

@xeffyr @fornwall do you want to run any particular tests or can we merge this and resolve possible problems afterwards?

**Comment by buttaface** [comment]
I'd like to help get this merged.

@Grimler91, if you're working on this now, let me know which packages have not been tried with this pull yet or that are breaking, and I will try building or fixing those. Let me know how I can chip in.

File: scripts/setup-ubuntu.sh [hunk 2]
```diff
@@ -107,9 +108,6 @@ PACKAGES+=" zlib1g-dev"
 PACKAGES+=" libssl-dev:i386"
 PACKAGES+=" zlib1g-dev:i386"
 
-# So we don't build llvm for build.
-PACKAGES+=" llvm-8-tools"
-
 # For swift.
 PACKAGES+=" clang-10"
```
Comment (buttaface) [inline scripts/setup-ubuntu.sh:112]: Redundant, can be removed.

**Comment by Grimler91** [comment]
@buttaface great! I have verified that all packages that are part of this PR builds without error. I am currently running a `./build-all.sh` build now just to check so that there are not a lot of other packages failing after the r21b->d update. 

I have not tested clang and rust on device after the update, have you tested them already? If not, could you run some simple compilation tests on device? 
I have uploaded the debs that are part of this PR built for aarch64 here: 
https://grimler.se/dists/testing/r21/ (it is possible to subscribe to the repository)

**Comment by buttaface** [comment]
>I have not tested clang and rust on device after the update

I've been using clang 10 and the full LLVM suite of packages built with those commits against NDK 20 on my device since I submitted the LLVM 10 pull months ago. It works great but I haven't tried it with NDK 21, though I doubt that will matter.

I've never used Rust on any platform, will try out your package.

**Comment by buttaface** [comment]
Tried your clang and rustc packages, dynamically linked against libc++ 20 and my previously built LLVM 10, and each was able to compile "hello world" in C++ and Rust just fine. Let me know if I should try anything else.

**Comment by Grimler91** [comment]
@buttaface nice, and the build-all.sh test-build is going well. I'll wait and see so that the CI does not fail due to build error on any of the other arches and will then merge this

**Comment by buttaface** [comment]
Looking forward to it. :tada:

**Comment by Grimler91** [comment]
I'm building the packages locally for all arches now since build time is too long for CI

**Comment by Grimler91** [comment]
rust for x86_64 fails with 

```
[...]
   Compiling mdbook-linkcheck v0.5.0
   Compiling rustbook v0.1.0 (/home/builder/.termux-build/rust/src/src/tools/rustbook)
    Finished release [optimized] target(s) in 2m 22s
Rustbook (x86_64-linux-android) - unstable-book
/home/builder/.termux-build/rust/build/build/x86_64-unknown-linux-gnu/stage0-tools-bin/rustbook: error while loading shared libraries: libc.so: ELF load command address/offset not properly aligned


command did not execute successfully: "/home/builder/.termux-build/rust/build/build/x86_64-unknown-linux-gnu/stage0-tools-bin/rustbook" "build" "/home/builder/.termux-build/rust/build/build/x86_64-linux-android/md-doc/unstable-book" "-d" "/home/builder/.termux-build/rust/build/build/x86_64-linux-android/doc/unstable-book"
expected success, got: exit code: 127


failed to run: /home/builder/.termux-build/rust/build/build/bootstrap/debug/bootstrap install --stage 2 --host x86_64-linux-android --target x86_64-linux-android --target wasm32-unknown-unknown
Build completed unsuccessfully in 0:19:47
```

So I'll blacklist x86_64 for now to be able to merge this

**Comment by buttaface** [comment]
:rocket:

**Pull request merged**